### DISCUSSION TITLE
玩转简悦专属收藏助手 - Telegram bot

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#2792] [Developer]
> 基于简悦的 API 实现的 Telegram bot 可以方便的将**任何 URL 发送到简悦的稍后读**中。

## 优点

> 得益于 Telegram 全平台方案，可以在 iOS / Android / Windows / Mac 方便的通过 Telegram bot 收集 URL。

## 一站式教程

https://www.yuque.com/kenshin/simpread/ieu1w6

使用一站式教程后，请根据 [此链接](https://www.yuque.com/kenshin/simpread/ieu1w6#AD2ZX) 开通相应权限。

## 获取 Token

通过上个步骤可以登录到 [开放平台](http://simpread.pro/developer)，然后复制截图所示的 Token

![image](https://user-images.githubusercontent.com/81074/132090944-2ce5b56e-bb9c-4a8e-b97b-01f5b1112515.png)

如果 Token 不存在可通过下图设置

![image](https://user-images.githubusercontent.com/81074/132090973-3bfee066-4a7e-49f5-bf8c-032bffbc1349.png)

### 添加 Telegram bot

> 点击 [@simpread_bot](https://t.me/simpread_bot) 即可添加。

## 设置 Token

![image](https://user-images.githubusercontent.com/81074/132091073-5924598c-5522-420d-8a4b-918fb670fe47.png)

> 输入 `/token xxxxx`  xxxx 为上个步骤复制到剪切板的内容，如： `/token 123456`

![image](https://user-images.githubusercontent.com/81074/132091129-50da33c5-42db-4811-9dbb-99c4343b7bfa.png)

## 功能

1. 支持任意 URL 的保存

2. 支持常见社交媒体的保存，细节请看 [使用 Telegram bot 玩转社交媒体的保存](https://github.com/Kenshin/simpread/discussions/3519)

3. 支持多组 URLs 的保存（每行一个）

    ![image](https://user-images.githubusercontent.com/81074/157815837-7f091a68-4632-4954-b1ea-0123e6596a21.png)

4. 当 URL 的快照识别失败时，确保将 URL 保存到你的稍后读（但没有快照，并提示失败的 URL）

    ![image](https://user-images.githubusercontent.com/81074/157815901-4497ac46-7867-41f6-abfc-7b0bbb3dff64.png)

## API 快照

支持 API 快照，如何配置 [请看这里](https://www.yuque.com/kenshin/simpread/ieu1w6#EmXOg)。

### 注意

> 当出现失败时，请看下面的解决方案：

- https://github.com/Kenshin/simpread/discussions/2919

- https://github.com/Kenshin/simpread/discussions/2399

- https://github.com/Kenshin/simpread/discussions/2281

- https://github.com/Kenshin/simpread/discussions/2282

## 如何使用

> 给 Telegram bot 发送任何包含链接的内容（包括转发），都会保存到你的 `坚果云 → SimpRead → simpread_config.json` 里面，如下图

![image](https://user-images.githubusercontent.com/81074/132091306-0cb548c6-45d8-4a7f-b0c3-9c5c60a16894.png)

## 如何同步到扩展端

> 上述操作后只能更新到  `坚果云 → SimpRead → simpread_config.json` 里面，需要使用下面的方式才能同步到你的浏览器简悦里面。

### 自动同步方案

>  使用 [安装和配置同步助手](https://github.com/Kenshin/simpread/discussions/2754)  提供的教程即可实现自动同步。

### 手动同步方案

> 需要手动同步，位置在：选项页 → 共通 → 手动同步

![image](https://user-images.githubusercontent.com/81074/132091443-5529bf10-37fe-46f7-ac00-cc50dc2c0c4a.png)

> 注意：上图使用的是强制覆盖方案，如果本地有个稍后读 A，但未同步到坚果云，而通过 Telegram bot 添加的 B 将会覆盖 A（也就是操作后 A 丢失）

### 综上

> 使用 Telegram bot 的前提是最好使用 **自动同步方案**，这样能获得更好的使用体验。

**Comment by hellozgl** [comment]
这个功能挺好用的，请问能否在技术上实现：通过telegram bot自动化功能，自动在稍后读中下载对应网页的html文件，类似于知识库的自动化。
另有个问题，发现simpread_config.json文件中包含了很多摘录内容。这个配置文件大小，会不会随着简悦的使用年限变长而变得非常庞大，有可能优化吗。

**Reply by Kenshin** [reply]
1. 支持 API 快照，如何配置 [请看这里](https://github.com/Kenshin/simpread/discussions/2994)。

2. 不会，目前我有 1915个稍后读数据，`simpread_config.json` 是 `11 mb`，对于坚果云或其它网盘来说，约等于秒传。全部数据在一个文件的好处是：方便转存和「携带」

**Reply by hellozgl** [reply]
谢谢。
请问网页url通过telegram bot，抑或通过简悦收藏助手来推送到稍后读列表，这两种方式产生的页面样式/页面精简度有区别吗？因为不了解这两个方式对网页处理的内在原理。
是不是相对来说，url发送给telegram小二，产生的稍后读内容会更加精简清爽一些呢？
另不知能否在稍后读中，对网页进行进一步的精简，live editor无法在稍后读里使用，其他的精简功能，也只能在网页阅读模式中进行，不能在稍后读中进行精简。

**Reply by Kenshin** [reply]
> 请问网页url通过telegram bot，抑或通过简悦收藏助手来推送到稍后读列表，这两种方式产生的页面样式/页面精简度有区别吗？

没有关系，虽然「前端界面」不一样，但都是调用相同的后端程序，你可以理解为 **快照只是一个服务器端 API，谁都能调用。**

> 在稍后读中，对网页进行进一步的精简，live editor无法在稍后读里使用，其他的精简功能，也只能在网页阅读模式中进行，不能在稍后读中进行精简。

这个「优化」会在下个版本中有所改善，暂时只能在阅读模式中使用 Live Editor 的方式实现。

**Reply by hellozgl** [reply]
刚刚又试了一下，如果使用telegram bot，可以一些网站的页面准确推送到稍后读中，并能够准确地下载好离线html。
但如果使用收藏助手，会显示“添加失败”，稍后读能接收到一个空的文件，没有离线html，但可以在阅读模式下打开该网页。从这个角度上来讲，貌似应该是telegram小二的通用性更强。而阅读助手只对国内网站比较友好。
![屏幕截图 2022-02-11 15 53 39](https://user-images.githubusercontent.com/70676660/153555901-84f8deb4-b584-4584-ad97-df68e8555b5d.png)
![屏幕截图 2022-02-11 15 53 55](https://user-images.githubusercontent.com/70676660/153555909-20a64080-33a8-4c0a-9342-1b8d9b9ed88e.png)
![屏幕截图 2022-02-11 15 55 34](https://user-images.githubusercontent.com/70676660/153555913-3ddb707b-bbe4-48a6-b1c8-6d285161247c.png)

**Reply by Kenshin** [reply]
> 刚刚又试了一下，如果使用telegram bot，可以将一些“海外”网站的页面准确推送到稍后读中，并能够准确地下载好离线html。
但如果使用收藏助手，会显示“添加失败”，稍后读能接收到一个空的文件，没有离线html，但可以在阅读模式下打开该网页（可能是有v2ray协助）。从这个角度上来讲，貌似应该是telegram小二的通用性更强。而阅读助手只对国内网站比较友好。

哈哈，基本上就是这样的，我统一回复在 https://github.com/Kenshin/simpread/discussions/3007#discussioncomment-2155566

**Comment by HSUCHING** [comment]
我试了一下,telegram显示发送成功,但是稍后读里面并没有.

**Reply by Kenshin** [reply]
要收藏到你的稍后读，需要：坚果云客户端 + 同步助手的自动同步

你可以看看坚果云 Web 端里面的 `simpread_config.json` 是否更新了。

**Reply by HSUCHING** [reply]
我看过了,更新了~ 但是还是不行. 另外,问个问题,我如果开启了同步助手,我的那个自动化流程好像就用不了了是吧?我看说明好像是2选1.

**Reply by Kenshin** [reply]
> 但是还是不行

那就是我说的 **坚果云客户端 + 同步助手的自动同步** 的问题，关于自动同步可以看 [这个教程](https://kb.simpread.pro/#/page/61f3d233-2dd8-40db-8daa-55f9dfdf6a7e)

> 开启了同步助手,我的那个自动化流程

并没有此限制 😂，建议详细说明，跟 Telegram bot 无关的问题，请通过 https://github.com/kenshin/simpread/issues/new 讨论。
